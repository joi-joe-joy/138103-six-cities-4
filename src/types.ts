export enum HouseType {
  APARTMENT = `apartment`,
  ROOM = `room`,
  HOUSE = `house`,
  HOTEL = `hotel`
};

export enum PlaceCardType {
  CITIES = `cities`,
  NEAR = `near`,
  FAVORITES = `favorites`,
  PROPERTY = `property`
};

export enum PageType {
  PROPERTY = `property`,
  MAIN = `main`,
  LOGIN = `login`,
  FAVORITES = `favorites`,
  FAVORITES_EMPTY = `favorites_empty`
};

export enum SortType {
  POPULAR = `popular`,
  TO_HIGH = `to-high`,
  TO_LOW = `to-low`,
  TOP_RATED = `top-rated`
};

export enum AuthStatus {
  NO_AUTH = `NO_AUTH`,
  AUTH = `AUTH`,
};

export interface Location {
  latitude: number,
  longitude: number,
  zoom: number
}

export interface City {
  location: Location,
  name: string
}

export interface User {
  id: number,
  avatarUrl: string,
  name: string,
  isPro: boolean
}

export type AuthInfo = User & {
  email: string
}

export interface Offer {
  id: number,
  title: string,
  maxAdults: number,
  description: string,
  isPremium: boolean,
  isFavorite: boolean,
  images: Array<string>,
  previewImage: string,
  goods: Array<string>,
  price: number,
  rating: number,
  bedrooms: number,
  type: HouseType.APARTMENT | HouseType.ROOM | HouseType.HOUSE | HouseType.HOTEL,
  host: User,
  location: Location,
  city: City,
};

export interface Comment {
  comment: string,
  date: string,
  id: number,
  rating: number,
  user: User
}
