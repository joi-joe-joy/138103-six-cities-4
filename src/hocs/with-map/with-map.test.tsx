import * as React from "react";
import * as renderer from "react-test-renderer";
import withMap from "./with-map";
import {Location, PageType} from "../../types";

const MockComponent = () => {
  return <div/>;
};

const MockComponentWrap = withMap(MockComponent);

const location: Location = {
  latitude: 45,
  longitude: 54,
  zoom: 12
};

it(`should render withMap correctly`, () => {

  const tree = renderer.create(
      <MockComponentWrap
        type={PageType.MAIN}
        offersCords={[location]}
        currentCords={location}
        cityLocation={location}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
