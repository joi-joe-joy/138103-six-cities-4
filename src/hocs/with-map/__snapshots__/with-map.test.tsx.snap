// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render withMap correctly 1`] = `<div />`;
